### DISCUSSION TITLE
Week 2 assignment

### DISCUSSION THREAD

**Discussion by sraorao** [sraorao/MSD_R_course_MT2021#8] [Show and tell]
Q1. Install the stemHypoxia data package from Bioconductor by following the instructions here: https://bioconductor.org/packages/devel/data/experiment/html/stemHypoxia.html

Q2. Load the stemHypoxia gene expression data.frame and draw a boxplot of gene expressions for all samples
hint: find the help page for the dataset to get you started. BUT, there is an error in the example code, how can you fix it?

Q3. Make a new column in the gene expression data.frame for the mean expression of each gene across all the samples. Sort the data.frame from highest to lowest mean expression and subset the top 10 genes (with the highest mean expression).
hint: Useful functions for this task are rowMeans(), base::order()/dplyr::arrange()

**Comment by yinyanxu** [comment]
Hi, I have a problem installing the package. I first tried to copy the installation code but later realize it's for R 4.2. I then clicked into 'Bioconductor release' page and chose R 4.1. Then I am not sure what to do but tried to search for the package 'BiocManager' but can't find it. 
So I tried to follow the error msg in R that I got after I ran the installation code for the wrong version. In the end I am stuck with these error messages: 

> BiocManager::install(version='devel')
Error: Bioconductor version '3.15' requires R version '4.2'; use `BiocManager::install(version = '3.14')` with R
  version 4.1; see https://bioconductor.org/install
> BiocManager::install(version = '3.14')
'getOption("repos")' replaces Bioconductor standard repositories, see '?repositories' for details

replacement repositories:
    CRAN: https://cran.rstudio.com/

Bioconductor version 3.14 (BiocManager 1.30.16), R 4.1.2 (2021-11-01)
Old packages: 'backports', 'glue'
Update all/some/none? [a/s/n]: 

I tried to answer with 'a' and got the following error message:

          binary source needs_compilation
backports  1.3.0  1.4.0              TRUE

  Binaries will be installed
trying URL 'https://cran.rstudio.com/bin/windows/contrib/4.1/backports_1.3.0.zip'
Content type 'application/zip' length 105005 bytes (102 KB)
downloaded 102 KB

trying URL 'https://cran.rstudio.com/bin/windows/contrib/4.1/glue_1.5.1.zip'
Content type 'application/zip' length 171670 bytes (167 KB)
downloaded 167 KB

package ‘backports’ successfully unpacked and MD5 sums checked
package ‘glue’ successfully unpacked and MD5 sums checked
Warning: cannot remove prior installation of package ‘glue’
Warning: restored ‘glue’

The downloaded binary packages are in
	C:\Users\xuyin\AppData\Local\Temp\RtmpCA8N8c\downloaded_packages
Warning message:
In file.copy(savedcopy, lib, recursive = TRUE) :
  problem copying C:\Users\xuyin\Documents\R\win-library\4.1\00LOCK\glue\libs\x64\glue.dll to C:\Users\xuyin\Documents\R\win-library\4.1\glue\libs\x64\glue.dll: Permission denied

I also tried again by answer 'n' and got another error message:

Warning message:
package(s) not installed when version(s) same as current; use `force = TRUE` to re-install: 'stemHypoxia' 

I'm confused and stuck. Sorry for only posting this now but I didn't expect to be stuck at the first step. If it's too complicated to solve the problem I'll also show up earlier tomorrow and ask you by Zoom. Thank you!

**Reply by sraorao** [reply]
Hi, from the warning message it looks like you already have the stemHypoxia installed (perhaps you managed to install it on one of your attempts), could you try loading it?

**Reply by yinyanxu** [reply]
Oh yes you're right! Thank you!
